lib/internal/main/test_runner.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+const { ArrayFrom, Promise, SafeSet } = primordials;
+const {
+  prepareMainThreadExecution
+} = require('internal/bootstrap/pre_execution');
+const { spawn } = require('child_process');
+const { readdirSync, statSync } = require('fs');
+const console = require('internal/console/global');
+const {
+  codes: {
+    ERR_TEST_FAILURE,
+  },
+} = require('internal/errors');
+const test = require('internal/test_runner/harness');
+const { kSubtestsFailed } = require('internal/test_runner/test');
+const { doesPathMatchFilter } = require('internal/test_runner/utils');
+const { join, resolve } = require('path');
+const kFilterArgs = ['--test'];
+
+prepareMainThreadExecution(false);
+markBootstrapComplete();
+
+// TODO(cjihrig): Replace this with recursive readdir once it lands.
+function processPath(path, testFiles, userSupplied = false) {
+  const stats = statSync(path);
+
+  if (stats.isFile()) {
+    if (userSupplied || doesPathMatchFilter(path)) {
+      testFiles.add(path);
+    }
+  } else if (stats.isDirectory()) {
+    if (!userSupplied && path.endsWith('node_modules')) {
```

Comment:
```suggestion
    if (!userSupplied && StringPrototypeEndsWith(path, 'node_modules')) {
```

Reply:
Same comment here about not seeing the benefit to primordials in this particular file.

Reply:
That's a fair point.